src/app/warehouse/warehouse.cpp

```diff
@@ -1,6 +1,20 @@
 #include "warehouse.hpp"
+#include "clinic_facade.hpp"
 #include "product.hpp"
 
+void Warehouse::createWarehouse()
+{
+    if (!Clinic::getWarehouse())
+    {
+        Warehouse warehouse;
```

Comment:
Same remark as for createCalendar.
Why create local warehouse just to copy it seconds later to heap?
Isn't it possible to create shared_ptr right away?
I didn't check, perhaps due to private constructor it's blocked.
The only reason that comes to me at the moment.